device/sycl/src/clusterization/clusterization_algorithm.sycl

```diff
@@ -9,45 +9,325 @@
 #include "../utils/get_queue.hpp"
```

Comment:
Do we have this code hooked up to our CCA tests? Would be useful to see whether this code does what it's supposed to do. I trust that it's a faithful translation but I am worried about subtle differences in memory and execution models between CUDA and SYCL.

Reply:
This would definately be good but if I understand correctly we currently don't have a way of executing SYCL code in our CI tests. I can however double check this by doing some further testing locally.